tests/components/notify/test_init.py

```diff
@@ -97,3 +143,197 @@ async def test_warn_template(hass, caplog):
     # We should only log it once
     assert caplog.text.count("Passing templates to notify service is deprecated") == 1
     assert hass.states.get("persistent_notification.notification") is not None
+
+
+async def test_invalid_platform(hass, caplog, tmp_path):
+    """Test service setup with an invalid platform."""
+    mock_notify_platform(hass, tmp_path, "testnotify1")
+    # Setup the platform
+    await async_setup_component(
+        hass, "notify", {"notify": [{"platform": "testnotify1"}]}
+    )
+    await hass.async_block_till_done()
+    assert "Invalid notify platform" in caplog.text
+    caplog.clear()
+    # Setup the second testnotify2 platform dynamically
+    mock_notify_platform(hass, tmp_path, "testnotify2")
+    await async_load_platform(
+        hass,
+        "notify",
+        "testnotify2",
+        {},
+        hass_config={"notify": [{"platform": "testnotify2"}]},
+    )
+    await hass.async_block_till_done()
+    assert "Invalid notify platform" in caplog.text
+
+
+async def test_invalid_service(hass, caplog, tmp_path):
+    """Test service setup with an invalid service object or platform."""
+
+    def get_service(hass, config, discovery_info=None):
+        """Return None for an invalid notify service."""
+        return None
+
+    mock_notify_platform(hass, tmp_path, "testnotify", get_service=get_service)
+    # Setup the second testnotify2 platform dynamically
+    await async_load_platform(
+        hass,
+        "notify",
+        "testnotify",
+        {"notify": [{"platform": "testnotify"}]},
```

Comment:
Please update the discovery info of all the load platform calls.

Reply:
done